monkey/monkey_island/cc/repositories/plugin_utils.py

```diff
@@ -0,0 +1,88 @@
+import json
+from tarfile import TarFile, TarInfo
+from typing import Any, BinaryIO, Dict
+
+import yaml
+
+from common.agent_plugins import AgentPlugin, AgentPluginManifest
+
+MANIFEST_FILENAME = "plugin.yaml"
+CONFIG_SCHEMA_FILENAME = "config-schema.json"
+SOURCE_ARCHIVE_FILENAME = "data.tar"
+
+
+def tarinfo_type(tar: TarInfo) -> str:
+    if tar.isfile():
+        return "file"
+    elif tar.isdir():
+        return "dir"
+    elif tar.issym():
+        return "symlink"
+    elif tar.islnk():
+        return "hardlink"
+    elif tar.isdev():
+        return "device"
+    return "unknown"
+
+
+def load_plugin(file: BinaryIO) -> AgentPlugin:
+    """
+    Load a plugin from a tar file.
+
+    :raises ValueError: if the file is not a valid plugin
+    """
+    tar_file = TarFile(fileobj=file)
+    try:
+        manifest = get_plugin_manifest(tar_file)
+        schema = get_plugin_schema(tar_file)
+        source = get_plugin_source(tar_file)
+    except KeyError as e:
+        raise ValueError("Bad plugin") from e
+
+    return AgentPlugin(plugin_manifest=manifest, config_schema=schema, source_archive=source)
+
+
+def get_plugin_manifest(tar: TarFile) -> AgentPluginManifest:
+    """
+    Retrieve the plugin manifest from a tar file.
+
+    :raises KeyError: if the manifest was not found in the tar file
+    :raises ValueError: if the manifest is not a file
+    """
+    manifest_info = tar.getmember(MANIFEST_FILENAME)
+    manifest_buf = tar.extractfile(manifest_info)
+    if not manifest_buf:
```

Comment:
Are `io.BufferedReader` objects falsey? I'm not sure. It's probably safest to do `if manifest_buf is None` everywhere.